### PR TITLE
clientv3: only start lease stream after first keepalive call

### LINKED ISSUE #7274: clientv3: don't open lease keepalive stream by default
Client will start lease stream even if no keep alives have been issued. Probably OK to keep the stream open after the first keepalive, but not to great opening a stream if it's never used (e.g., `etcdctl get`).

### PR DESCRIPTION
**Pull request by heyitsanthony**
Fixes #7274

### CODE REVIEW

**Comment by fanminshi** [comment]
lgtm defer to @xiang90

**Comment by heyitsanthony** [comment]
Failing on unrelated issue #7283. Jumping the turnstile and merging.

File: clientv3/lease.go [hunk 1]
```diff
@@ -279,6 +285,7 @@ func (l *lessor) KeepAliveOnce(ctx context.Context, id LeaseID) (*LeaseKeepAlive
 
 func (l *lessor) Close() error {
 	l.stopCancel()
+	l.firstKeepAliveOnce.Do(func() { close(l.donec) })
```
Comment (xiang90) [inline clientv3/lease.go]: this seems to be tricky. I feel we should document this at least.

Comment (heyitsanthony) [inline clientv3/lease.go]: ok, fair enough

**Comment by xiang90** [comment]
LGTM

**Pull request merged**